src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -609,7 +614,14 @@ bool GetPeerNodeInfo(CommandHandler * commandHandler, const ConcreteCommandPath
 
     auto strongDelegate = mDelegate;
     dispatch_async(mDelegateNotificationQueue, ^{
-        if ([strongDelegate respondsToSelector:@selector(handleQueryImageForNodeID:controller:params:completion:)]) {
+        if ([strongDelegate respondsToSelector:@selector(handleQueryImageForNodeID:
+                                                                        controller:params:canStartNewBDXSession:completion:)]) {
+            [strongDelegate handleQueryImageForNodeID:@(nodeId)
+                                           controller:controller
+                                               params:commandParams
+                                canStartNewBDXSession:gOtaSender.CanStartNewBDXSession()
```

Comment:
So if `canStartNewBDXSession` here is false, what is the callee expected to do?

Reply:
if canStartNewBDXSession is false, the callee will send query image available otherwise if the flag is true, we will return a resource busy since the sdk is busy with another BDX session.

Reply:
Then why does the callee need to know that?  Can't the SDK itself return the "Busy" status if it's busy, without bothering the delegate at all?

Reply:
The answer to my question above is that the logic needs to be:

* If no image available, return "not available"
* Else if busy return "busy"
* Else return success.

That first condition needs to be evaluated by the delegate.

Given that, I would propose the following API:

* Add a `bdxTransferState:(MTRBDXTransferState *)bdxTransferState` to the selector/args.
* Define a `MTRBDXTransferState` interface, which for now just has a single BOOL property called canStartNewBDXSession.

Then later:

1. We can consider exposing the progress of the various active BDX sessions in the transfer state, in case the callee wants to cancel some of the "not far enough along" ones to service a higher-priority update, if we start supporting canceling sessions.
2. If we start supporting more than one session, we can communicate how many are in flight now.

etc, etc. without changing the basic API.

Thoughts?